### PR TITLE
AKO-1.12.1 Release Notes

### PR DESCRIPTION
**Pull request by akshayhavile**

### CODE REVIEW

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Comment by akshayhavile** [comment]
skip jira-id-check

File: CHANGELOG.md [hunk 1]
```diff
@@ -445,3 +445,28 @@ All notable changes to this project will be documented in this file. The format
...
+
+### Added
+- AKO now claims support for Kubernetes 1.29, OCP version 4.14.
+- AKO supports LoadBalancerClass for LoadBalancer type services.
+- AKO allows pool member to be IPV6 or IPV4 in dual stack deployment for `CALICO` and `ANTREA` CNI.
+- IPV6 address can be used to connect to the Avi Controller.
+- VRF support within VCenter Cloud
+- AKO now has a `L7Rule` CRD to change default parameters of L7 VirtualService in addition to `Hostrule` CRD.
+- Support for Network Security policy in HostRule CRD.
+- Allow AKO to use existing avi-secret in avi-system namespace to connect to Avi Controller.
```
Comment (DixitAakash) [inline CHANGELOG.md:459]: Can we change `avi-system` to `AKO Helm release` namespace.

File: CHANGELOG.md [hunk 2]
```diff
@@ -445,3 +445,28 @@ All notable changes to this project will be documented in this file. The format
...
+
+## AKO-1.12.1
+
+### Added
+- AKO now claims support for Kubernetes 1.29, OCP version 4.14.
+- AKO supports LoadBalancerClass for LoadBalancer type services.
+- AKO allows pool member to be IPV6 or IPV4 in dual stack deployment for `CALICO` and `ANTREA` CNI.
+- IPV6 address can be used to connect to the Avi Controller.
+- VRF support within VCenter Cloud
+- AKO now has a `L7Rule` CRD to change default parameters of L7 VirtualService in addition to `Hostrule` CRD.
```
Comment (pkoshtavmware) [inline CHANGELOG.md:457]: Hostrule-> "HostRule"

File: CHANGELOG.md [hunk 3]
```diff
@@ -445,3 +445,28 @@ All notable changes to this project will be documented in this file. The format
...
+- Support for Network Security policy in HostRule CRD.
+- Allow AKO to use existing avi-secret in avi-system namespace to connect to Avi Controller.
+
+### Changed
+- L4 CRD can be applied to LB type service present in different namespace.
```
Comment (pkoshtavmware) [inline CHANGELOG.md:462]: LB -> LoadBalancer

File: CHANGELOG.md [hunk 4]
```diff
@@ -445,3 +445,28 @@ All notable changes to this project will be documented in this file. The format
...
+- AKO shows all IPV4 and IPV6 addresses associated with VirtualService as part of Ingress, LB status.
+- AKO updates `RouteStatus:Reason` field with vsuuid and controller uuid.
+
+### Fixed
+- AKO allows VIP to be IPV6 for LB type services.
```
Comment (pkoshtavmware) [inline CHANGELOG.md:467]: LB -> LoadBalancer

File: CHANGELOG.md [hunk 5]
```diff
@@ -445,3 +445,28 @@ All notable changes to this project will be documented in this file. The format
...
+- Fix: Static routes are not added due to an error: `Unique constraints route_id has duplicated value when calico CNI allocates multiple block-affinities for a node` and that may result in AKO crash.
+- Fix: Failure in creating Virtual service, PG and other pools if name of one of the pool, attached to PG, exceeds maximum avi controller length limit.
```
Comment (pkoshtavmware) [inline CHANGELOG.md:469]: length limit -> name length limit ?
the pool -> the Pool
avi controller -> Avi Controller
Maybe 🤷‍♀️

File: CHANGELOG.md (hunk 2 above)
Comment (pkoshtavmware) [reply CHANGELOG.md:457]: Have we missed adding multiport support for services here ?

File: CHANGELOG.md [hunk 6]
```diff
@@ -445,3 +445,28 @@ All notable changes to this project will be documented in this file. The format
...
+- Allow AKO to use existing avi-secret in avi-system namespace to connect to Avi Controller.
+
+### Changed
+- L4 CRD can be applied to LB type service present in different namespace.
+- AKO shows all IPV4 and IPV6 addresses associated with VirtualService as part of Ingress, LB status.
+- AKO updates `RouteStatus:Reason` field with vsuuid and controller uuid.
+
+### Fixed
+- AKO allows VIP to be IPV6 for LB type services.
+- Fix: Static routes are not added due to an error: `Unique constraints route_id has duplicated value when calico CNI allocates multiple block-affinities for a node` and that may result in AKO crash.
```
Comment (pkoshtavmware) [inline CHANGELOG.md:468]: Can we rewrite it as "Fix: Static Routes are not added due to an error: `Field check for static_routes failed : Unique constraints route_id has duplicated value`, when Calico CNI allocates multiple block-affinities for a node that may result in AKO crash"

File: CHANGELOG.md (hunk 2 above)
Comment (akshayhavile) [reply CHANGELOG.md:457]: @pkoshtavmware Can you please give me link for PR `Multiport support for Services` ?

Comment (pkoshtavmware) [reply CHANGELOG.md:457]: https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pull/1373/files.
My bad, it was only for Gateways not ingresses

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Comment by akshayhavile** [comment]
skip jira-id-check

**Review by pkoshtavmware — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 7]
```diff
@@ -445,3 +445,28 @@ All notable changes to this project will be documented in this file. The format
...
+- AKO shows all IPV4 and IPV6 addresses associated with VirtualService as part of Ingress, LoadBalancer status.
+- AKO updates Route's `Status:Reason` field with vsuuid and controller uuid.
+
+### Fixed
+- AKO allows VIP to be IPV6 for LoadBalancer type services.
+- Fix: Static routes are not added due to an error: `Field check for static_routes failed: Unique constraints route_id has duplicated value` when calico CNI allocates multiple block-affinities for a node and that may result in AKO crash.
+- Fix: Failure in creating Virtual Service, PoolGroup and other Pools if name of one of the Pool, attached to PoolGroup, exceeds maximum Avi Controller name length limit.
+
+### Known Issue
+- L4 Rule CRD, with `enableSSL: true` in listener propterties, will not be applied to L4 Virtual Service if Avi Controller license is of type `Enterprise with Cloud Service`.
```
Comment (DixitAakash) [inline CHANGELOG.md:472]: @akshayhavile should this be `license is not of type` ?

Comment (akshayhavile) [reply CHANGELOG.md:472]: Only Enterprise and Enterprise with Cloud Service can create L4 SSL. As per my understanding, for others ... there is no way they can create L4 SSL VS

Comment (DixitAakash) [reply CHANGELOG.md:472]: @akshayhavile So you mean only `Enterprise` and `Enterprise with Cloud Service' license types are allowed to create L4 SSL VS? And due to the restriction in our code only `Enterprise` will be actually allowed to do it?

**Review by DixitAakash — APPROVED** [review]

**Review by arihantg — APPROVED** [review]

**Pull request merged**